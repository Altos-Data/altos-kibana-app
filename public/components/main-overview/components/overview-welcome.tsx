/*
 * Wazuh app - React component for building the Overview welcome screen.
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React from 'react';
import StringTools from '../../../utils/strings-tools';
import {
  EuiCard,
  EuiIcon,
  EuiPanel,
  EuiFlexItem,
  EuiFlexGroup,
  EuiSpacer,
  EuiFlexGrid,
  EuiCallOut,
  EuiPage,
  EuiButton,
} from '@elastic/eui';
import { updateCurrentTab } from '../../../redux/actions/appStateActions';
import store from '../../../redux/store';
import './welcome.scss';
import { TabDescription } from '../../../../server/reporting/tab-description';

export const OverviewWelcome = (props) => {
  const buildTabCard = (tab, icon) => {
    return (
      <EuiFlexItem>
        <EuiCard
          layout="horizontal"
          icon={<EuiIcon size="xl" type={icon} color="primary" />}
          className="homSynopsis__card"
          title={TabDescription[tab].title}
          onClick={() => store.dispatch(updateCurrentTab(tab))}
          data-test-subj={`overviewWelcome${StringTools.capitalize(tab)}`}
          description={TabDescription[tab].description}
        />
      </EuiFlexItem>
    );
  };

  const addAgent = () => {
    return (
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiCallOut
            style={{ height: '65%' }}
            title="No agents were added to this manager. "
            color="warning"
            iconType="alert"
          >
            <EuiButton style={{ margin: '-58px 286px' }} color="warning" href="#/agents-preview?">
              Add agent
            </EuiButton>
          </EuiCallOut>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  return (
    <EuiPage className="wz-welcome-page">
      <EuiFlexGroup>
        <EuiFlexItem>
          {props.agentsCountTotal == 0 && addAgent()}
          <EuiFlexGroup>
            <EuiFlexItem>
              <EuiPanel betaBadgeLabel="Security Information Management">
                <EuiSpacer size="s" />
                <EuiFlexGrid columns={2}>
                  {buildTabCard('general', 'dashboardApp')}
                  {buildTabCard('fim', 'filebeatApp')}
                  {props.extensions.aws && buildTabCard('aws', 'logoAWSMono')}
                  {props.extensions.gcp && buildTabCard('gcp', 'logoGCPMono')}
                </EuiFlexGrid>
              </EuiPanel>
            </EuiFlexItem>
            <EuiFlexItem>
              <EuiPanel betaBadgeLabel="Auditing and Policy Monitoring">
                <EuiSpacer size="s" />
                <EuiFlexGrid columns={2}>
                  {buildTabCard('pm', 'advancedSettingsApp')}
                  {props.extensions.audit && buildTabCard('audit', 'monitoringApp')}
                  {props.extensions.oscap && buildTabCard('oscap', 'codeApp')}
                  {props.extensions.ciscat && buildTabCard('ciscat', 'auditbeatApp')}
                  {buildTabCard('sca', 'securityAnalyticsApp')}
                </EuiFlexGrid>
              </EuiPanel>
            </EuiFlexItem>
          </EuiFlexGroup>

          <EuiSpacer size="xl" />
          <EuiFlexGroup>
            <EuiFlexItem>
              <EuiPanel betaBadgeLabel="Threat Detection and Response">
                <EuiSpacer size="s" />
                <EuiFlexGrid columns={2}>
                  {buildTabCard('vuls', 'securityApp')}
                  {props.extensions.virustotal && buildTabCard('virustotal', 'savedObjectsApp')}
                  {props.extensions.osquery && buildTabCard('osquery', 'searchProfilerApp')}
                  {props.extensions.docker && buildTabCard('docker', 'logoDocker')}
                  {buildTabCard('mitre', 'spacesApp')}
                  {/* TODO- Change "spacesApp" icon*/}
                </EuiFlexGrid>
              </EuiPanel>
            </EuiFlexItem>

            <EuiFlexItem>
              <EuiPanel betaBadgeLabel="Regulatory Compliance">
                <EuiSpacer size="s" />
                {!props.extensions.pci &&
                  !props.extensions.gdpr &&
                  !props.extensions.hipaa &&
                  !props.extensions.tsc &&
                  !props.extensions.nist && (
                    <EuiFlexGroup>
                      <EuiFlexItem>
                        <EuiCallOut
                          title={
                            <p>
                              Click the <EuiIcon type="eye" /> icon to show regulatory compliance
                              extensions.
                            </p>
                          }
                          color="success"
                          iconType="help"
                        />
                      </EuiFlexItem>
                    </EuiFlexGroup>
                  )}
                {(props.extensions.pci ||
                  props.extensions.gdpr ||
                  props.extensions.hipaa ||
                  props.extensions.tsc ||
                  props.extensions.nist) && (
                  <EuiFlexGrid columns={2}>
                    {props.extensions.pci && buildTabCard('pci', 'visTagCloud')}
                    {props.extensions.nist && buildTabCard('nist', 'apmApp')}
                    {props.extensions.tsc && buildTabCard('tsc', 'apmApp')}
                    {props.extensions.gdpr && buildTabCard('gdpr', 'visBarVertical')}
                    {props.extensions.hipaa && buildTabCard('hipaa', 'emsApp')}
                  </EuiFlexGrid>
                )}
              </EuiPanel>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiPage>
  );
};
