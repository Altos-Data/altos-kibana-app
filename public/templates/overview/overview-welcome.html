<md-content flex layout="column" ng-if="tab === 'welcome'" layout-align="start">

    <div layout="row" class="md-padding font-size-18">
        <span flex>
            <i class="fa fa-fw fa-server"></i> Host monitoring
        </span>
    </div>

    <div layout="row" layout-align="start start" layout-wrap>
        <md-card flex class="wz-md-card">
            <md-card-content>
                <i class="fa fa-fw fa-home"></i> <span ng-click="switchTab('general')" class="wz-headline-title wz-text-link cursor-pointer">General</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>Brief overview of your latest Wazuh alerts</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/user-manual/capabilities/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
        <md-card flex class="wz-md-card">
            <md-card-content>
                <i class="fa fa-fw fa-file"></i> <span ng-click="switchTab('fim')" class="wz-headline-title wz-text-link cursor-pointer">File Integrity</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>Alerts from monitored files and directories</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/user-manual/capabilities/file-integrity/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
        <md-card flex class="wz-md-card" ng-show="extensions.aws">
            <md-card-content>
                <i class="fa fa-fw fa-amazon"></i> <span ng-click="switchTab('aws')" class="wz-headline-title wz-text-link cursor-pointer">AWS</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>AWS CloudTrail logs from AWS S3 buckets</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/amazon/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
    </div>

    <div layout="row" class="md-padding font-size-18">
        <span flex>
            <i class="fa fa-fw fa-flag"></i> System audit
        </span>
    </div>

    <div layout="row" layout-align="start start" layout-wrap>
        <md-card flex class="wz-md-card">
            <md-card-content>
                <i class="fa fa-fw fa-desktop"></i> <span ng-click="switchTab('pm')" class="wz-headline-title wz-text-link cursor-pointer">Policy Monitoring</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>Anomaly detection of unexpected behavior</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/user-manual/capabilities/policy-monitoring/rootcheck/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
        <md-card flex class="wz-md-card" ng-show="extensions.audit">
            <md-card-content>
                <i class="fa fa-fw fa-balance-scale"></i> <span ng-click="switchTab('audit')" class="wz-headline-title wz-text-link cursor-pointer">Audit</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>Linux Audit system alerts</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/user-manual/capabilities/system-calls-monitoring/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
        <md-card flex class="wz-md-card" ng-show="extensions.oscap">
            <md-card-content>
                <i class="fa fa-fw fa-search"></i> <span ng-click="switchTab('oscap')" class="wz-headline-title wz-text-link cursor-pointer">Open SCAP</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>Alerts about OpenSCAP security policies</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/user-manual/capabilities/policy-monitoring/openscap/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
    </div>

    <div layout="row" class="md-padding font-size-18">
        <span flex>
            <i class="fa fa-fw fa-shield"></i> Security
        </span>
    </div>

    <div layout="row" layout-align="start start" layout-wrap>
        <md-card flex class="wz-md-card">
            <md-card-content>
                <i class="fa fa-fw fa-bug"></i> <span ng-click="switchTab('vuls')" class="wz-headline-title wz-text-link cursor-pointer">Vulnerabilities</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>Alerts about known vulnerabilities on your applications</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/user-manual/capabilities/vulnerability-detection.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
        <md-card flex class="wz-md-card" ng-show="extensions.virustotal">
            <md-card-content>
                <i class="fa fa-fw fa-file-o"></i> <span ng-click="switchTab('virustotal')" class="wz-headline-title wz-text-link cursor-pointer">VirusTotal</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>Alerts from VirusTotal malicious content scans</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/user-manual/capabilities/virustotal-scan/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
    </div>

    <div layout="row" class="md-padding font-size-18" ng-show="extensions.pci || extensions.gdpr">
        <span flex>
            <i class="fa fa-fw fa-check"></i> Compliance
        </span>
    </div>

    <div layout="row" layout-align="start start" layout-wrap ng-show="extensions.pci || extensions.gdpr">
        <md-card flex class="wz-md-card" ng-show="extensions.pci">
            <md-card-content>
                <i class="fa fa-fw fa-credit-card"></i> <span ng-click="switchTab('pci')" class="wz-headline-title wz-text-link cursor-pointer">PCI DSS</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>PCI DSS compliance alerts</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/pci-dss/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
        <md-card flex class="wz-md-card" ng-show="extensions.gdpr">
            <md-card-content>
                <i class="fa fa-fw fa-list-ol"></i> <span ng-click="switchTab('gdpr')" class="wz-headline-title wz-text-link cursor-pointer">GDPR</span>
                <md-divider class="wz-margin-top-10"></md-divider>
                <div layout="row" class="wz-padding-top-10">
                    <p>GDPR compliance alerts</p>
                </div>
            </md-card-content>
            <md-card-actions layout="row" layout-align="end center">
                <md-button target="_blank" href="https://documentation.wazuh.com/current/gdpr/index.html" class="wz-button-flat small">
                    <i class="fa fa-fw fa-info"></i> More info
                </md-button>
            </md-card-actions>
        </md-card>
    </div>

</md-content>
