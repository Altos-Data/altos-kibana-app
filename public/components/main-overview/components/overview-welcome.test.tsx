/*
 * Wazuh app - React component for building the Overview welcome screen - Test
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { OverviewWelcome } from './overview-welcome';

describe('Overview welcome component', () => {
  test('should render a Overview welcome screen', () => {
    const extensions = {
      audit: true,
      pci: true,
      gdpr: true,
      hipaa: true,
      nist: true,
      tsc: true,
      oscap: true,
      ciscat: true,
      aws: true,
      gcp: true,
      virustotal: true,
      osquery: true,
      docker: true,
    };
    const component = shallow(<OverviewWelcome extensions={extensions} />);

    expect(component).toMatchSnapshot();
  });

  test('should render a Overview welcome screen with any extension', () => {
    const extensions = {
      audit: false,
      pci: false,
      gdpr: false,
      hipaa: false,
      nist: false,
      tsc: false,
      oscap: false,
      ciscat: false,
      aws: false,
      gcp: false,
      virustotal: false,
      osquery: false,
      docker: false,
    };
    const component = shallow(<OverviewWelcome extensions={extensions} />);

    expect(component).toMatchSnapshot();
  });
});
