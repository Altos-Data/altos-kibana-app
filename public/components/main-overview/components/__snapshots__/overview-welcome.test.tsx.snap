// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview welcome component should render a Overview welcome screen 1`] = `
<EuiPage
  className="wz-welcome-page"
>
  <EuiFlexGroup>
    <EuiFlexItem>
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Security Information Management"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGrid
              columns={2}
            >
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeGeneral"
                  description="Browse through your security alerts, identifying issues and threats in your environment."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="dashboardApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Security events"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeFim"
                  description="Alerts related to file changes, including permissions, content, ownership and attributes."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="filebeatApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Integrity monitoring"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeAws"
                  description="Security events related to your Amazon AWS services, collected directly via AWS API."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="logoAWSMono"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Amazon AWS"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeGcp"
                  description="Security events related to your Google Cloud Platform services, collected directly via GCP API."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="logoGCPMono"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Google Cloud Platform"
                />
              </EuiFlexItem>
            </EuiFlexGrid>
          </EuiPanel>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Auditing and Policy Monitoring"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGrid
              columns={2}
            >
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomePm"
                  description="Verify that your systems are configured according to your security policies baseline."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="advancedSettingsApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Policy monitoring"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeAudit"
                  description="Audit users behavior, monitoring command execution and alerting on access to critical files."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="monitoringApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="System auditing"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeOscap"
                  description="Configuration assessment and automation of compliance monitoring using SCAP checks."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="codeApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="OpenSCAP"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeCiscat"
                  description="Configuration assessment using Center of Internet Security scanner and SCAP checks."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="auditbeatApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="CIS-CAT"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeSca"
                  description="Scan your assets as part of a configuration assessment audit."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="securityAnalyticsApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Security configuration assessment"
                />
              </EuiFlexItem>
            </EuiFlexGrid>
          </EuiPanel>
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer
        size="xl"
      />
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Threat Detection and Response"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGrid
              columns={2}
            >
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeVuls"
                  description="Discover what applications in your environment are affected by well-known vulnerabilities."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="securityApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Vulnerabilities"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeVirustotal"
                  description="Alerts resulting from VirusTotal analysis of suspicious files via an integration with their API."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="savedObjectsApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="VirusTotal"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeOsquery"
                  description="Osquery can be used to expose an operating system as a high-performance relational database."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="searchProfilerApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Osquery"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeDocker"
                  description="Monitor and collect the activity from Docker containers such as creation, running, starting, stopping or pausing events."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="logoDocker"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Docker listener"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeMitre"
                  description="Security events from the knowledge base of adversary tactics and techniques based on real-world observations"
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="spacesApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="MITRE ATT&CK"
                />
              </EuiFlexItem>
            </EuiFlexGrid>
          </EuiPanel>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Regulatory Compliance"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGrid
              columns={2}
            >
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomePci"
                  description="Global security standard for entities that process, store or transmit payment cardholder data."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="visTagCloud"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="PCI DSS"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeNist"
                  description="National Institute of Standards and Technology Special Publication 800-53 (NIST 800-53) sets guidelines for federal information systems."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="apmApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="NIST 800-53"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeTsc"
                  description="Trust Services Criteria for Security, Availability, Processing Integrity, Confidentiality, and Privacy"
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="apmApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="TSC"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeGdpr"
                  description="General Data Protection Regulation (GDPR) sets guidelines for processing of personal data."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="visBarVertical"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="GDPR"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeHipaa"
                  description="Health Insurance Portability and Accountability Act of 1996 (HIPAA) provides data privacy and security provisions for safeguarding medical information."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="emsApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="HIPAA"
                />
              </EuiFlexItem>
            </EuiFlexGrid>
          </EuiPanel>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlexItem>
  </EuiFlexGroup>
</EuiPage>
`;

exports[`Overview welcome component should render a Overview welcome screen with any extension 1`] = `
<EuiPage
  className="wz-welcome-page"
>
  <EuiFlexGroup>
    <EuiFlexItem>
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Security Information Management"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGrid
              columns={2}
            >
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeGeneral"
                  description="Browse through your security alerts, identifying issues and threats in your environment."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="dashboardApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Security events"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeFim"
                  description="Alerts related to file changes, including permissions, content, ownership and attributes."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="filebeatApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Integrity monitoring"
                />
              </EuiFlexItem>
            </EuiFlexGrid>
          </EuiPanel>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Auditing and Policy Monitoring"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGrid
              columns={2}
            >
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomePm"
                  description="Verify that your systems are configured according to your security policies baseline."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="advancedSettingsApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Policy monitoring"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeSca"
                  description="Scan your assets as part of a configuration assessment audit."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="securityAnalyticsApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Security configuration assessment"
                />
              </EuiFlexItem>
            </EuiFlexGrid>
          </EuiPanel>
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer
        size="xl"
      />
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Threat Detection and Response"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGrid
              columns={2}
            >
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeVuls"
                  description="Discover what applications in your environment are affected by well-known vulnerabilities."
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="securityApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="Vulnerabilities"
                />
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiCard
                  className="homSynopsis__card"
                  data-test-subj="overviewWelcomeMitre"
                  description="Security events from the knowledge base of adversary tactics and techniques based on real-world observations"
                  icon={
                    <EuiIcon
                      color="primary"
                      size="xl"
                      type="spacesApp"
                    />
                  }
                  layout="horizontal"
                  onClick={[Function]}
                  title="MITRE ATT&CK"
                />
              </EuiFlexItem>
            </EuiFlexGrid>
          </EuiPanel>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiPanel
            betaBadgeLabel="Regulatory Compliance"
          >
            <EuiSpacer
              size="s"
            />
            <EuiFlexGroup>
              <EuiFlexItem>
                <EuiCallOut
                  color="success"
                  iconType="help"
                  title={
                    <p>
                      Click the 
                      <EuiIcon
                        type="eye"
                      />
                       icon to show regulatory compliance extensions.
                    </p>
                  }
                />
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiPanel>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlexItem>
  </EuiFlexGroup>
</EuiPage>
`;
